File: Submit/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i, v in enumerate(nums):
            search = map.get((target - v), None)
            if search is not None:
                return [search, i]

            map[v] = i


# @lc code=end

File: Submit/1.two-sum.rs
/*
 * @lc app=leetcode id=1 lang=rust
 *
 * [1] Two Sum
 */

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;

        let mut map: HashMap<i32, i32> = HashMap::with_capacity(nums.len());
        for (i, v) in nums.iter().enumerate() {
            match map.get(&(target - v)) {
                Some(&i2) => return vec![i2, i as i32],
                None => map.insert(*v, i as i32),
            };
        }
        vec![]
    }
}
// @lc code=end

